### ISSUE TITLE
About errors when training in Resnet with CIFAR10

### ISSUE THREAD

**Issue by dzfowen** [tensorflow/models#3612]
Hi, when I tried  models/research/resnet/  there is a error:

Caused by op 'SparseToDense', defined at:
  File "/Users/dzf/models/research/resnet/resnet_main.py", line 215, in <module>
    tf.app.run()
  File "/anaconda/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 48, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "/Users/dzf/models/research/resnet/resnet_main.py", line 207, in main
    train(hps)
  File "/Users/dzf/models/research/resnet/resnet_main.py", line 55, in train
    FLAGS.dataset, FLAGS.train_data_path, hps.batch_size, FLAGS.mode)
  File "/Users/dzf/models/research/resnet/cifar_input.py", line 109, in build_input
    [batch_size, num_classes], 1.0, 0.0)
  File "/anaconda/lib/python3.6/site-packages/tensorflow/python/ops/sparse_ops.py", line 557, in sparse_to_dense
    name=name)
  File "/anaconda/lib/python3.6/site-packages/tensorflow/python/ops/gen_sparse_ops.py", line 1032, in _sparse_to_dense
    validate_indices=validate_indices, name=name)
  File "/anaconda/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 768, in apply_op
    op_def=op_def)
  File "/anaconda/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2336, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/anaconda/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1228, in __init__
    self._traceback = _extract_stack()

InvalidArgumentError (see above for traceback): indices[0] = [0,180] is out of bounds: need 0 <= index < [128,10]
	 [[Node: SparseToDense = SparseToDense[T=DT_FLOAT, Tindices=DT_INT32, validate_indices=true, _device="/job:localhost/replica:0/task:0/cpu:0"](concat, SparseToDense/output_shape, SparseToDense/sparse_values, SparseToDense/default_value)]]

By stackoverflow, it seems a problem of CIFAR10 and its classes maybe not from 0 to 9. I tried to reconstruct label matrix by minus 1 but helpless. So what is wrong with this piece of codes.

Thanks a lot.

**Comment by parallelo** [comment]
I believe I saw this as well.  I think dropping back to an older commit helped.  For example: 
```
cd models
git checkout -b oct2017 fb12693f2420cc687d0ca8c640c241610b2b036b
```
If that fixes your issue, perhaps try a `git-bisect` to find the faulty commit.

**Comment by dzfowen** [comment]
@parallelo sorry  did you mean resnet located at /models/official/resnet?

Because I tried /models/research/resnet as your instructions but failed. It seems it has no this branch.

Thanks a lot.

**Comment by parallelo** [comment]
That'll create a local branch based on a specific commit and switch you to it.  
```
git checkout -b <branch-name> <sha1-of-commit>
```

**Comment by karmel** [comment]
The version of ResNet at research/resnet is the responsibility of the original researcher to maintain, and we encourage you to instead take a look at the Official ResNet (official/resnet), which includes a maintained version of ResNet for both ImageNet and Cifar10.
